### PR TITLE
Added a grid style ui for browsing games on the games page

### LINKED ISSUE #96: Update game list view to be a grid of game covers/posters
We will work on making a game list view for the games in the Chigame database. The listview will include the name of the game and a cover photo.

### PR DESCRIPTION
**Pull request by willcravitz**
This allows users to browse through games as cards that display the game image, title, and short description. These cards link out to their respective game detail pages. The new template is named `game_grid.html` and I've kept the old `game_list.html` just in case we need a list view in the future.

### CODE REVIEW

File: src/templates/games/game_grid.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  </style>
+  <h1 class="display-1 mt-3 mb-3">Games</h1>
+  <div class="row row-cols-1 row-cols-md-4 g-4">
+    {% for game in object_list %}
+      <div class="col">
+        <div class="card h-100">
+          <img src="{{ game.image }}" class="card-img-top" alt="game image" />
+          <div class="card-body">
+            <h4 class="card-title">{{ game.name }}</h4>
+            <p class="card-text">{{ game.description }}</p>
```
Comment (ptlin25) [inline src/templates/games/game_grid.html]: Add `|linebreaksbr` to game.description so that the game description properly displays new lines

**Review by ptlin25 — CHANGES_REQUESTED** [review]
New lines in the game description are not properly displayed. See the comment on the code

File: src/templates/games/game_grid.html (hunk 1 above)
Comment (willcravitz) [reply src/templates/games/game_grid.html]: Thanks for calling this out, just pushed a fix.

**Review by ptlin25 — CHANGES_REQUESTED** [review]
I think the game images/thumbnail should be the same height, so that the game titles and descriptions are aligned.
Screenshot 2023-11-16 at 10 08 00 AM

**Comment by frowenz** [comment]
> I think the game images/thumbnail should be the same height, so that the game titles and descriptions are aligned. Screenshot 2023-11-16 at 10 08 00 AM

Relevant CSS for this is [`object-fit: contain`](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit)

**Comment by frowenz** [comment]
### Changes made:
1. Images are [contained](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) with a fixed area. Behind a given image is a blurred background image that [covers]
(https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) the area. Hence, this fixes the previous description problem.
2. All cards now have the same height.
3. Previously, there was a set number (4) of cards per row so the cards would get very skinny. The design is now more responsive.
4. Removed footer and added direct linking from the card.

### Future work
1. Descriptions will go on as long as they need. We should cut them off at some point.
2. The mobile view looks very bad.

### VIdeo
https://github.com/uchicago-cs/chigame/assets/69015620/9cabe83e-cb06-4c0e-ac39-513c8261216f

**Review by ptlin25 — APPROVED** [review]
I agree that the description needs to be capped in the future. Otherwise, good job addressing these issues! The grid view looks amazing!

**Review by majorsylvie — APPROVED** [review]
Looks absolutely incredible! 
I enjoy how they have the reactive size change upon mouse-over.
Zooming in and out a lot looks a little funny, but still is adaptive and doesn't make it impossible to use.

Next step is now to have the search bar a gridview page like this, instead still leading to a hyperlink list.
Or if you still want a list view, making more comprehensive cards that also have the image, short description, etc...

The small bubbles for categories is a very nice touch :)

**Comment by frowenz** [comment]
In this pull request, we developed a page featuring a grid layout to showcase games.

This involved crafting a template page enhanced with nice CSS. Both @frowenz and @willcravitz worked on this page. My contribution was the initial creation of the basic grid layout, while @willcravitz enhanced it by integrating bootstrap cards for a more visually appealing. I made some final adjustments to these cards, such as better responsiveness and consistent positioning of the titles.

Looking ahead, there are a couple of areas for improvement:

The descriptions currently have no length limit, which needs to be addressed.
The page's mobile responsiveness is currently suboptimal and requires enhancement.

By merging this pull request, we will also be resolving issue #96.

**Comment by majorsylvie** [comment]
Issue Score: Excellent

Comments:
Amazing work all of you!!

@ptlin25 thank you for your good and persistent PR review, making sure this great feature get's merged with multiple improvements made.

@frowenz @willcravitz Thank you for your phenomenal work creating and styling the template. 
Wonderful closing comment owen.
Great summary will!

Thank you all for your work. Y'all deserve to be proud of what you've accomplished.

**Pull request merged**